platforms/ios/src/TangramMap/TGMarker.mm

```diff
@@ -7,244 +7,145 @@
 //
 
 #import "TGMarker.h"
+#import "TGMarker+Internal.h"
 #import "TGMapViewController.h"
 #import "TGMapViewController+Internal.h"
 #import "TGHelpers.h"
+#import "TGTypes.h"
 #import "tangram.h"
 
-enum class TGMarkerType {
-    point,
-    polygon,
-    polyline,
-    none,
-};
-
 @interface TGMarker () {
     Tangram::Map* tangramInstance;
-    Tangram::MarkerID identifier;
-    TGMarkerType type;
+    NSError* error;
 }
 
-@property (copy, nonatomic) NSString* stylingString;
-@property (copy, nonatomic) NSString* stylingPath;
-@property (assign, nonatomic) TGGeoPoint point;
-@property (strong, nonatomic) TGGeoPolyline* polyline;
-@property (strong, nonatomic) TGGeoPolygon* polygon;
-@property (assign, nonatomic) BOOL visible;
-@property (assign, nonatomic) NSInteger drawOrder;
-@property (strong, nonatomic) UIImage* icon;
-@property (weak, nonatomic) TGMapViewController* map;
-
-- (void)createNSError:(NSError **)error;
+- (void)createNSError;
+- (void)clearGeometry;
 
 @end
 
 @implementation TGMarker
 
-- (instancetype)init
+- (instancetype)initWithMap:(Tangram::Map*)map
 {
     self = [super init];
 
     if (self) {
-        type = TGMarkerType::none;
+        tangramInstance = map;
+        self.identifier = tangramInstance->markerAdd();
         self.visible = YES;
         self.drawOrder = 0;
     }
 
     return self;
 }
 
-- (instancetype)initWithMapView:(TGMapViewController *)mapView
+- (void)clearGeometry
 {
-    self = [super init];
-
-    if (self) {
-        type = TGMarkerType::none;
-
-        self.visible = YES;
-        self.drawOrder = 0;
-
-        [self map:mapView error:nil];
-    }
-
-    return self;
+    self.polyline = nil;
+    self.polygon = nil;
+    self.point = {NAN, NAN};
 }
 
-- (void)createNSError:(NSError **)error
+- (void)createNSError
 {
-    if (!error) {
-        return;
-    }
+    // TODO: Check for all Tangram internal tangram error and
```

Comment:
I am not sure I understand the "Check for all Tangram internal error" here. Shouldn't `createNSError` receive some context of the error from the callee to be used to get this additional info?

Or if tangram core is maintaining a stack or error, does this grab the last error and use it to populate the additional info?